### PR TITLE
Fixed: Build failed due to "old-style function"

### LINKED ISSUE #122: Carthage fails to archive project
macOs Big Sur 11.4
Xcode 12.5 (Intel)
Aardvark/4.0.0

`
Carthage/Checkouts/Aardvark/Sources/CoreAardvark/Logging/ARKExceptionLogging.m:36:52: error: this old-style function definition is not preceded by a prototype [-Werror,-Wstrict-prototypes]
NSLock * ARKGetUncaughtExceptionLogDistributorsLock()
`

[Carthage-aardvark.log](https://github.com/square/Aardvark/files/6557948/Carthage-aardvark.log)

### PR DESCRIPTION
**Pull request by destryteeter**
Resolves #122

### CODE REVIEW

File: Sources/CoreAardvark/Logging/ARKExceptionLogging.m [hunk 1]
```diff
@@ -33,7 +33,7 @@
 ARKUncaughtExceptionHandlerGetter ARKGetUncaughtExceptionHandler = NSGetUncaughtExceptionHandler;
 ARKUncaughtExceptionHandlerSetter ARKSetUncaughtExceptionHandler = NSSetUncaughtExceptionHandler;
 
-NSLock * ARKGetUncaughtExceptionLogDistributorsLock()
+NSLock * ARKGetUncaughtExceptionLogDistributorsLock(void)
```
Comment (dfed) [inline Sources/CoreAardvark/Logging/ARKExceptionLogging.m:36]: I think you'll need to put this newer style declaration behind an `#if` statement checking that we're using a later Xcode version. Right now Aardvark still supports Xcode 11, so we can't land changes that break that version.

**Review by dfed — COMMENTED** [review]
Thanks for letting us know about the issue and putting up a PR! Still a little more work to do though: we gotta get CI green.

File: Sources/CoreAardvark/Logging/ARKExceptionLogging.m (hunk 1 above)
Comment (destryteeter) [reply Sources/CoreAardvark/Logging/ARKExceptionLogging.m:37]: Thanks, I included another commit that applies this last fix for Xcode 12 an newer

**Comment by destryteeter** [comment]
Any chance this can get merged in?

**Review by dfed — APPROVED** [review]
This LGTM, but it's been awhile since I've done much with compiler declarations in Objective-C, and I no longer work on an app that uses this repo.

@NickEntin can you PTAL and take this over the line?

**Review by NickEntin — APPROVED** [review]
LGTM, thanks for fixing this! Apologies for the delayed response, this slipped through my review queue.

Not sure why the CI build isn't getting kicked off. I'll try closing and reopening this PR to see if that triggers it.

Also, have you [signed our CLA](https://github.com/square/Aardvark/blob/master/Contributing.md#sign-the-cla)?

**Comment by destryteeter** [comment]
Thanks, just signed the CLA.

**Pull request merged**